post/clang_tidy_review/clang_tidy_review/__init__.py

```diff
@@ -847,21 +855,56 @@ def save_metadata(pr_number: int) -> None:
         json.dump(metadata, metadata_file)
 
 
-def load_review() -> Optional[PRReview]:
-    """Load review output from the standard REVIEW_FILE path.
-    This file contains
+def load_review(review_file) -> Optional[PRReview]:
+    """Load review output"""
 
-    """
-
-    if not pathlib.Path(REVIEW_FILE).exists():
-        print(f"WARNING: Could not find review file ('{REVIEW_FILE}')", flush=True)
+    if not pathlib.Path(review_file).exists():
+        print(f"WARNING: Could not find review file ('{review_file}')", flush=True)
         return None
 
-    with open(REVIEW_FILE, "r") as review_file:
-        payload = json.load(review_file)
+    with open(review_file, "r") as review_file_handle:
+        payload = json.load(review_file_handle)
         return payload or None
 
 
+def load_and_merge_reviews(review_files) -> Optional[PRReview]:
+    reviews = []
+    for file in review_files:
+        review = load_review(file)
+        if review is not None:
+            reviews.append(review)
+
+    if not reviews:
+        return None
+
+    result = reviews[0]
+
+    class Comment:
+        def __init__(self, data):
+            self.data = data
+
+        def __hash__(self):
+            return hash(
+                (
+                    self.data["body"],
+                    self.data["line"],
+                    self.data["path"],
+                    self.data["side"],
+                )
+            )
+
+        def __eq__(self, other):
+            return type(other) is Comment and self.data == other.data
+
+    comments = OrderedSet()
```

Comment:
What does `OrderedSet` give us over- plain `set`? Why do we need the comments to be ordered?

Reply:
When merging the comments, I thought it would be more deterministic if the order wasn't changed but you're right that it's not needed. Shall I remove it?

Reply:
That's a good point. Maybe could just replace it with a `set` and then do `sorted(list(comments))`?

Reply:
What will it sort on?

Reply:
Oh of course, it uses insertion order, like `OrderedDict`. On balance though, I think I would still prefer not to use a whole extra dependency just for this.

I think clang-tidy sorts them by filename then line number? I'm not sure beyond that.